unit/size_left_test.c

```diff
@@ -141,12 +141,16 @@ rx_size_left_err(void)
 		goto fail;
 	}
 
-	/* ep starts in a non-enabled state, may fail, should not SEGV */
-	fi_rx_size_left(ep);
+	/* ep starts in a non-enabled state, so if supported this
+	 * should return -FI_EOPBADSTATE */
+	ret = fi_rx_size_left(ep);
+	if ((ret != -FI_EOPBADSTATE) && (ret != -FI_ENOSYS)) {
+		printf("fi_rx_size_left %s (%d)\n", fi_strerror(-ret), ret);
+		goto fail;
+	}
 
 	testret = PASS;
 fail:
-	FT_CLOSE_FID(ep);
 	return TEST_RET_VAL(ret, testret);
```

Comment:
Thanks, @jithinjosepkl.  I see that now.  I must have misread the valgrind output.  I swore it said no memory leaks, but I'll take another (closer) look.